Message送信機能のビュー

avaklee opened the pull request:
# What
(1) nullでもsaveできるようにdbの書き直し(2) 完成品を参考にroutingの書き直し(3) view file/scss の整理

# Why
見た目の完成度を上げるため、user flowの再定義が必要だったため

avaklee committed:
Message送信機能のビュー

# What
1) nullでもsaveできるようにdbの書き直し2) 完成品を参考にroutingの書き直し
3) view file/scss の整理
# Why
見た目の完成度を上げるため、

yusukeyamane commented on app/controllers/messages_controller.rb:
```diff
@@ -18,4 +21,8 @@ def message_params
     params.require(:message).permit(:body, :image)
   end
 
+  def grouping
```
メソッド名は動詞にしましょう

yusukeyamane commented on app/views/groups/index.html.haml:
```diff
@@ -8,20 +8,27 @@
+          .left_content__chat_groups__chat_group_name
+            "#{group.name}"
+            .left_content__chat_groups__last_msg
+              - if group.messages.present?
```
これ以下はかなり長い文章になってしまうので、viewではなくhelperにまとめて書いてしまいましょう！

yusukeyamane commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,48 @@
+          -@group.users.each do |f|
+            "Members: #{f.name}"
+      .right_content__middle_content
+        - if @messages.present?
```
この部分は部分テンプレートにして呼び出せば、if文を使用する必要がなくなると思います。

yusukeyamane commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,48 @@
+              .right_content__middle_content__message__comment
+                = msg.body
+
+      = form_for @message, :url => {:action => :create} do |f|
```
url: 適切なprefixという形で書き直しましょう

avaklee committed:
Message送信機能のビュー

# What
html.hamlを一部部分テンプレートに移行、ビュー上に書かれていたruby codeをmethod化、method
nameのverb化、うurl- pathに変更
# Why
ビューのファイルを見やすく、かつフロントエンド側とのチームで進めるときに、エンジニアのコードはなるべくビューから離した方がフロント側が理解・作業
しやすいので。

seoky0 commented on app/assets/stylesheets/modules/_main_content.scss:
```diff
@@ -65,12 +65,13 @@
 
 
-          &__sent_button{
+          #sent_button{
```
なぜidにしたのでしょうか？

seoky0 replied:
また、sentではなく、sendだと思われます。

avaklee replied:
form forを利用したら、ヴュー画面で崩れてしまったためです。元のクラスに指定を変えます。

yusukeyamane commented on app/views/groups/index.html.haml:
```diff
@@ -6,29 +6,18 @@
-            .right_content__middle_content__message__comment
-            = msg.body
-
+        = render partial: "messages", locals: { messages: @messages }
```
こちら
render @messages
と書くだけで同じ事が実現できると思います。かなり省略できますね！

seoky0 commented on app/controllers/messages_controller.rb:
```diff
@@ -1,15 +1,18 @@
 
   def create
-    @message = Message.new(message_params)
+    @message = Message.create(body: message_params[:body], image:message_params[:image], group_id: params[:group_id], user_id: current_user.id)
```
message_paramsのようなメソッドを定義して、Message.new(message_params)とできるようにしましょう。

seoky0 replied:
また、createだと保存失敗時の処理が書けないので、newとsaveを分けて使いましょう。

avaklee committed:
Message送信機能のビュー

# WHAT
(1)グループを新規作成後、行き先をグループのindexではなく、メッセージのインデックスに変更
(2) paramsの書き方変更、new and saveで分ける
(3) viewのadjustment

# WHY
createだと保存失敗時の処理が書けないので分ける。後は見た目・簡略化のため

seoky0 commented on app/views/groups/index.html.haml:
```diff
@@ -6,22 +6,9 @@
-        .right_content__bottom_content__text_bar
-        Please enter text
+          .left_content__chat_groups__chat_group_name
+            #{show_group_name(group)}
```
= show_group_name(group)
と書けるはずです。
式展開は、式の結果 + 文字列のように文字列をつなげたいときに使用するものなので、式の結果だけを表示したいのであれば、= を使いましょう。

avaklee committed:
最後の修正

式展開をより
